src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.Unix.cs

```diff
@@ -81,111 +82,52 @@ private static Version GetICUVersion()
                                 ver >> 24);
         }
 
-        private static DistroInfo ParseOsReleaseFile()
+        private static DistroInfo GetDistroInfo() => new DistroInfo()
         {
-            Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.Linux));
+            Id = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystem,
+            VersionId = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystemVersion
```

Comment:
> VersionId = M [](start = 11, length = 14)

can we change this to Version struct instead of string? that will make it easier to use in the code and in comparison too,

Reply:
@tarekgh, @eerhardt sorry for late response here, I had to suspend this work for a while - I've just tried converting the code to use version but there is one detail which may cause us some problems and might be worth mentioning: Version class requires that we pass at minimum 2 arguments (major, minor) and does not allow negative minor version - in some cases we might expect something to work on any 6.X version (like with RedHat or CentOS case) and in some cases to work only on 6.0 but not on other 6.X versions - this might be problematic. I was thinking that example code comparing version could look something like following:
```csharp
// NOTE: this does not work as expected
        private static bool VersionEquivalentWith(Version expectedVersionId, Version actualVersionId)
        {
            return expectedVersionId.Major == actualVersionId.Major
                && expectedVersionId.Minor == actualVersionId.Minor
                && (expectedVersionId.Build == -1 || expectedVersionId.Build == actualVersionId.Build)
                && (expectedVersionId.Revision == -1 || expectedVersionId.Revision == actualVersionId.Revision);
        }
```

but this is causing problems as mentioned above (i.e. expecting RHEL6, got 6.9 => condition failed). We might approach working around this limitation in different ways:
- use some placeholder number as "minor not defined" i.e. 9999999
- compare ranges - that sounds fairly non-intuitive to define, i.e. checking for any RHEL6 would be something along the lines of: X >= V(6) && X < V(7), where more intuitive X == V(6) will fail
- define *expected* version in terms of 4 numbers and *actual* as Version class - that could work but not sure if that would make the code cleaner as all of these variable version parts would have to be handled
- leave string comparison

any opinions? Perhaps there is some way to simplify this overall.

Reply:
I think when we don't have a minor version we just use 0 which should work for any checks use this version. what would be wrong with that?

Reply:
@tarekgh that was the first thing I've tried. Example problem: You want condition like:
IsCentOS7 -> you have to compare with new Version(7,0) -> let's assume you're on CentOS 7.2 - should this comparison return true or false?
Other case: IsCentOS70, you check for new Version(7,0) -> no matter what you answered before this is a problematic situation

Reply:
Ok, talked with @tarekgh offline we will go with:
- parse to Version class in such a way that when minor is not provided interpret it as minor = 0
- each function which compares versions takes 4 integers and compares them with Version handling minor == -1